src/main/java/org/ethereum/beacon/discovery/schema/NonceGenerator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.ethereum.beacon.discovery.schema;
+
+import java.util.Random;
+import java.util.function.Function;
+import org.apache.tuweni.bytes.Bytes;
+import org.ethereum.beacon.discovery.type.Bytes12;
+
+public class NonceGenerator implements Function<Random, Bytes12> {
+  private static final int RANDOM_PART_SIZE = 8;
+
+  private int counterEnd = 0;
+  private int counter = 0;
+  private Bytes randomPart;
+
+  @Override
+  public synchronized Bytes12 apply(Random random) {
+    if (counter == counterEnd) {
+      randomPart = Bytes.random(RANDOM_PART_SIZE, random);
+      counter = random.nextInt();
+      counterEnd = counter;
+    }
+    counter++;
```

Comment:
So the aim here is to generate a new randomPart each time counter wraps the whole way around an int value? That works in terms of tracking uniqueness which I think is all we need but just wanting to double check I'm understanding this right.

Reply:
Yep, this is exactly what the spec recommends: 
> Since the wire protocol uses 96 bit AES-GCM nonces, it is strongly recommended to generate them by encoding the current outgoing message count into the first 32 bits of the nonce and filling the remaining 64 bits with random data generated by a cryptographically secure random number generator.